test/cfg/boost.cpp

```diff
@@ -104,4 +105,25 @@ void lock_guard_finiteLifetime(boost::mutex& m)
 {
     // cppcheck-suppress unusedScopedObject
     boost::lock_guard<boost::mutex>{ m };
-}
\ No newline at end of file
+}
+
+BOOST_AUTO_TEST_SUITE(my_auto_test_suite)
+
+BOOST_AUTO_TEST_CASE(test_message_macros)
+{
+    bool my_bool = false;
+    BOOST_WARN_MESSAGE(my_bool, "warn");
+    BOOST_CHECK_MESSAGE(my_bool, "check");
+    BOOST_REQUIRE_MESSAGE(my_bool, "require");
+
+    BOOST_WARN_MESSAGE(my_bool, "my_bool was: " << my_bool);
+}
+
+using test_types_w_tuples = std::tuple<int, long, unsigned char>;
+BOOST_AUTO_TEST_CASE_TEMPLATE(my_tuple_test, T, test_types_w_tuples)
```

Comment:
I don't remember how the `valueFlowBailoutIncompleteVar` is activated, is there a risk that regular users could get it?  I do not want that regular users will have to add such suppressions.

Reply:
`valueFlowBailoutIncompleteVar` is activated with the command line flag `--debug-warnings`. This is passed to Cppcheck in test/CMakeList.txt `add_cfg`. I don't think may users will pass this flag. The bad thing about this warning is that Cppcheck will probably not check the rest of this function.

Reply:
The checks will still run, just ValueFlow might skip some steps.
Is it possible to turn `BOOST_AUTO_TEST_CASE_TEMPLATE_run` into a function template with parameter `T`, or would this break something?

Reply:
I changed `BOOST_AUTO_TEST_CASE_TEMPLATE` to a function template now. I tried this change in our CI and there where no issues. However I couldn't remove the suppression for the `valueFlowBailoutIncompleteVar`.

Reply:
Yeah, that's probably because the template is not instantiated, see e.g. https://trac.cppcheck.net/ticket/12550